docs/recipes/stencil.md

```diff
@@ -0,0 +1,79 @@
+# Usage with Stencil
+
+[Stencil](https://stenciljs.com/) web components work very well with XState.
+
+### `src/helpers/toggle-machine.ts`
+
+```js
+import { createMachine } from "@xstate/fsm";
+
+export const toggleMachine = createMachine<{ toggle: "active" | "inactive" }>({
+  id: "toggle",
+  initial: "active",
+  states: {
+    inactive: { on: { toggle: "active" } },
+    active: { on: { toggle: "inactive" } }
+  }
+});
+```
+
+### `src/components/toggle/toggle.tsx`
+
+Add a `state` property to your component, decorated with `@State` so that it triggers a re-render when changed.
+
+On `componentWillLoad`, interpret the `toggleMachine` and listen for state transitions.
+
+After a transition has occured, the `state` property is set to the machine's new state, triggering a re-render.
+
+```js
+import { Component, h, State } from "@stencil/core";
+import { interpret } from "@xstate/fsm";
+import { toggleMachine } from "../helpers/toggle-machine";
+
+@Component({
+  tag: "my-toggle",
+  styleUrl: "toggle.css",
+  shadow: true
+})
+export class Toggle {
+  private _service = interpret(toggleMachine);
+
+  @State() state = toggleMachine.initialState;

```

Comment:
I tested this and it doesn't actually get the initial state for me... maybe it's better to just do

```suggestion
  @State() state: typeof toggleMachine.initialState;
```

Reply:
Hm that was also because of that caching issue I think, it does actually work to get the initial state!